### PR TITLE
Remove dependency on zap

### PR DESCRIPTION
**Pull request by jantytgat**
Replaced all references to zap.Logger with equivalents from slog.

### CODE REVIEW

File: examples/porcelain/main.go [hunk 1]
```diff
@@ -48,10 +49,7 @@ func highLevelExample() error {
 	ctx := context.Background()
 
 	// Logging is important - replace with your own zap logger
-	logger, err := zap.NewDevelopment()
-	if err != nil {
-		return err
-	}
+	logger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelDebug}))
```
Comment (francislavoie) [inline examples/porcelain/main.go:52]: This comment mentions zap

**Comment by francislavoie** [comment]
Looks pretty good to me, I think. But this is really @mholt's project so it's his call.

File: examples/porcelain/main.go (hunk 1 above)
Comment (jantytgat) [reply examples/porcelain/main.go]: Latest commit removes zap from the comments

**Review by mholt — APPROVED** [review]
Sorry for the delay. Been really behind after Thanksgiving.

Thanks for the PR -- looks good, I guess. I'm new to slog but at a glance it all looks fine, and should only be a minor performance regression overall.

`v3` is :melting_face: but I guess that's how it goes. Thank you!

**Pull request merged**